src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -129,12 +133,33 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
     /// </summary>
     /// <typeparam name="T">Type of container resource.</typeparam>
     /// <param name="builder">Builder for the container resource.</param>
-    /// <param name="image">Registry value.</param>
+    /// <param name="image">Image value.</param>
     /// <returns></returns>
     public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder, string image) where T : ContainerResource
     {
-        var containerImageAnnotation = builder.Resource.Annotations.OfType<ContainerImageAnnotation>().Single();
-        containerImageAnnotation.Image = image;
+        return builder.WithImage<T>(image, "latest");
+    }
+
+    /// <summary>
+    /// Allows overriding the image on a container.
+    /// </summary>
+    /// <typeparam name="T">Type of container resource.</typeparam>
+    /// <param name="builder">Builder for the container resource.</param>
+    /// <param name="image">Image value.</param>
+    /// <param name="tag">Tag value.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder, string image, string tag) where T : ContainerResource
```

Comment:
Do we need 2 overloads? Can't we just do tag= "latest" as a default argument?

Reply:
We surely can, I just tried to follow the same pattern used in the other extensions